File: src/139. Word Break.js
/**
 * @param {string} s
 * @param {string[]} wordDict
 * @return {boolean}
 */

export function wordBreak(s, wordDict) {
  const dp = Array(s.length + 1).fill(false)
  dp[0] = true
  for (let i = 0; i < dp.length; i++) {
    for (let j = 0; j < i; j++) {
      if (dp[j] === true && wordDict.includes(s.slice(j, i))) {
        dp[i] = true
        break
      }
    }
  }
  return dp[dp.length - 1]
}

File: src/139. Word Break.cpp
class Solution {
    public:
        bool wordBreak(string s, vector < string > & wordDict) {
            int n = s.size();
            vector < bool > dp(n + 1);
            dp[0] = true;
            for (int i = 1; i <= n; i++) {
                for (int j = 0; j < i; j++) {
                    if (dp[j] == true && find(wordDict.begin(), wordDict.end(), s.substr(j, i - j)) != wordDict.end()) {
                        dp[i] = true;
                        break;
                    }
                }
            }
            return dp.back();
        }
};